src/main/java/bookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -38,9 +40,36 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
     public ResponseEntity<Map<String, Object>> handleEntityNotFoundException(
             EntityNotFoundException ex
     ) {
+        return getCustomExceptionHandler(ex, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<Map<String, Object>> handleRegistrationException(
+            RegistrationException ex
+    ) {
+        return getCustomExceptionHandler(ex, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(BadCredentialsException.class)
+    public ResponseEntity<Map<String, Object>> handleLoginException(
+            BadCredentialsException ex
+    ) {
+        return getCustomExceptionHandler(ex, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(JwtException.class)
+    public ResponseEntity<Map<String, Object>> handleJwtException(
```

Comment:
have you tested this handler? Does it works?

Reply:
No, I actually didn't. And I wonder why now, it's a bit unclear to me

Reply:
because handler handles exceptions from controller, but jwtException throws in filter, so you can delete this handler